packages/webapp/src/routes/__tests__/credits-purchase/check-and-submit.spec.js

```diff
@@ -28,19 +34,14 @@ describe(url, () => {
       const res = await submitGetRequest({ url }, 200, session.values)
       expect(res.payload).not.toContain('Geoff')
     })
-
-    it('should handle missing values from the metric', async () => {
-      const session = setCreditsApplicationSession()
-      session.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_DATA, { startPage: { } })
-      await submitGetRequest({ url }, 200, session.values)
-    })
   })
 
   describe('POST', () => {
     it('should process a valid application correctly', done => {
       jest.isolateModules(async () => {
         try {
           const session = setCreditsApplicationSession()
+          populateSessionWithDevelopmentProjectInfo(session)
```

Comment:
Whilst this works as intended, it would be more elegant to update the mock session for credits purchase with these three values instead. Then you don't need to inject the session values here or later in `credits-application-validation.spec.js`.

The line above (`const session = setCreditsApplicationSession()`) reads in a whole mock session for credits purchase from here: https://github.com/DEFRA/biodiversity-net-gain-service/blob/master/packages/webapp/src/__mocks__/credits-application-data.js, so you can just add a value for `'credits-purchase-planning-authority-list'`, `'credits-purchase-planning-application-ref'`, and `'credits-purchase-planning-development-name'` to that file

Reply:
Nice thanks for the review, done